great_expectations/core/data_context_key.py

```diff
@@ -74,3 +79,25 @@ def to_fixed_length_tuple(self):
     @classmethod
     def from_fixed_length_tuple(cls, tuple_):
         return cls.from_tuple(tuple_)
+
+
+class DataContextVariableKey(DataContextKey):
+    def __init__(
+        self,
+        resource_type: DataContextVariableSchema,
+        resource_name: Optional[str] = None,
+    ) -> None:
+        self._resource_type = resource_type
+        self._resource_name = resource_name
```

Comment:
Some `DataContextVariables` are strings or other single values while other ones are dictionaries.

If I want to use my `InlineStoreBackend` to add a new datasource or store, I want the ability to specify my type (i.e. datasource or store) as well as the name associated with the new object (i.e. my_new_datasource or my_new_store).

The old `StringKey` did not allow for this flexibility.